extensions/mongodb/pom.xml

```diff
@@ -48,40 +48,39 @@
                 </configuration>
             </plugin>
 
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-failsafe-plugin</artifactId>
-                <version>${maven.failsafe.plugin.version}</version>
-                <configuration combine.children="append">
-                    <!--
-                    Failsafe uses the jar, but the main artifact, we shade the dependencies into a jar with jar-with-dependencies classifier
-                    Moreover this parameter name is really confusing.
-                    -->
-                    <classesDirectory>${project.build.directory}/${project.artifactId}-${project.version}-jar-with-dependencies.jar</classesDirectory>
-                </configuration>
-            </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-shade-plugin</artifactId>
                 <version>${maven.shade.plugin.version}</version>
+                <configuration>
+                    <shadedArtifactAttached>false</shadedArtifactAttached>
+                    <relocations>
+                        <relocation>
+                            <pattern>org.apache.calcite</pattern>
+                            <shadedPattern>${relocation.root}.org.apache.calcite</shadedPattern>
+                        </relocation>
+                    </relocations>
+                    <artifactSet>
+                        <includes>
+                            <include>org.apache.calcite:calcite-core</include>
+                        </includes>
+                    </artifactSet>
+                </configuration>
                 <executions>
                     <execution>
                         <id>jar-with-dependencies</id>
                         <phase>package</phase>
                         <goals>
                             <goal>shade</goal>
                         </goals>
-                        <configuration>
-                            <relocations>
-                                <relocation>
-                                    <pattern>org.apache.calcite</pattern>
-                                    <shadedPattern>${relocation.root}.org.apache.calcite</shadedPattern>
-                                </relocation>
-                            </relocations>
-                        </configuration>
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
```

Comment:
Do we need this entry, what's wrong with the config from the root pom.xml?

Reply:
TBH I don't exactly know why I added it, I tried many different options - I removed it as indeed it's not needed in the final version